File: solutions/typescript/grains/1/grains.ts
export const square = (n: number): bigint => {
  if (n < 1 || n > 64) {
    throw new Error('Square must be between 1 and 64');
  }
  return 2n ** BigInt(n - 1);
}

export const total = (): bigint => {
  return (2n ** 64n) - 1n;
}

File: solutions/julia/grains/1/grains.jl

function on_square(square)
    if square < 1 || square > 64
        throw(DomainError(square, "Square must be between 1 and 64"))
    end

    return Int128(1) << (square - 1)
end


function total_after(square)
    if square < 1 || square > 64
        throw(DomainError(square, "Square must be between 1 and 64"))
    end
    return (Int128(1) << square) - 1
end
